crates/ruff_python_formatter/src/expression/binary_like.rs

```diff
@@ -20,13 +20,182 @@ use crate::expression::string::StringLayout;
 use crate::expression::OperatorPrecedence;
 use crate::prelude::*;
 
-pub(super) struct BinaryLike<'a>(pub(super) &'a ExprBinOp);
+#[derive(Copy, Clone, Debug)]
+pub(super) enum BinaryLike<'a> {
+    BinaryExpression(&'a ExprBinOp),
+    CompareExpression(&'a ExprCompare),
+}
+
+impl<'a> BinaryLike<'a> {
+    fn flatten(
```

Comment:
This is mostly moved from `flatten_binary_expression` except that it now supports compare expressions too